In [72]:
# Importando las librerias básicas
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

**Eliminando datos atípicos y que carecen de sentido**
---

---


In [95]:
df=pd.read_csv("heart.csv")
# Como observamos anteriormente en los datos de colesterol se encontraba un grupo
# atípico así que revisaremos sus caracteristicas
print("El numero de valores diferentes en Cholesterol menores de 50 es:", df[df['Cholesterol']<50]['Cholesterol'].nunique())
print("La cantidad de datos menores de 50 es: ", df[df['Cholesterol']<50]['Cholesterol'].count(), "\n")
# Se encontró que los valores eran 0, no corresponden con valores reales por lo que 
# se asume que se trataron de valores perdidos.
# De igual forma en Oldpeak se encuentran algunos datos negativos
print("El numero de valores diferentes menores que 0 en Oldpeak es: ", df[df['Oldpeak']<0]['Oldpeak'].nunique())
print("La cantidad de datos menores de 50 es: ", df[df['Oldpeak']<0]['Oldpeak'].count())

El numero de valores diferentes en Cholesterol menores de 50 es: 1
La cantidad de datos menores de 50 es:  172 

El numero de valores diferentes menores que 0 en Oldpeak es:  10
La cantidad de datos menores de 50 es:  13


In [96]:
# Como los valores representaban menos del 20% de la data y aún eliminandolos
# tendriamos un volumen considerable para trabajar, aquí se elimina ese grupo
df=df[df['Cholesterol']>50]
# Similar para el caso de Oldpeak valores negativos parecen no tener un significado
# médico real por lo que se consideran outliers y se eliminan (son solo 13 datos).
df=df[df['Oldpeak']>=0]
df.describe().T

count        mean        std   min    25%    50%    75%    max
Age           745.0   52.906040   9.489625  28.0   46.0   54.0   59.0   77.0
RestingBP     745.0  133.036242  17.290451  92.0  120.0  130.0  140.0  200.0
Cholesterol   745.0  244.747651  59.113689  85.0  208.0  237.0  275.0  603.0
FastingBS     745.0    0.167785   0.373927   0.0    0.0    0.0    0.0    1.0
MaxHR         745.0  140.209396  24.536108  69.0  122.0  140.0  160.0  202.0
Oldpeak       745.0    0.902953   1.072953   0.0    0.0    0.5    1.5    6.2
HeartDisease  745.0    0.477852   0.499845   0.0    0.0    0.0    1.0    1.0

In [100]:
df.to_csv("df_heartfailure.csv")

**OBSERVACIONES**
---

---



1.   Un total de 173 datos fueron eliminados.
2.   El numero de instancias final fue de 745.
3.   Ya no se tienen datos que carezcan de sentido en las variables de Cholesterol y Oldpeak.
